### DISCUSSION TITLE
How to cancel an event in a trigger?

### DISCUSSION THREAD

**Discussion by ethndotsh** [supabase/supabase#6578] [Questions] [answered]
In Supabase, I know it is possible to create a trigger that fires before the event, but what if I want to add validation to that trigger before the event. How would I cancel the event if it does not meet my requirements?

Perhaps an RPC update function?

**Comment by GaryAustin1** [comment, marked as answer]
Your question is sort of unclear.
If you mean a Postgres before trigger on a table update or insert, then you return null instead of new to cancel the update/insert.
Not sure what RPC has to do with a trigger event other than it might do insert/update/delete to a table that then causes the trigger event above.
If you have an RPC function you can certainly do any tests you want before doing the Postgres SQL calls.

**Reply by ethndotsh** [reply]
Apologies for being unclear. I am really tired at the moment and apparently am not able to put together a coherent post.
